src/app/mods/health-bar/health-bar.ts

```diff
@@ -1,153 +1,148 @@
-                Logger.error(ex);
-            }
-        });
+    private showHealthBars(){
+        if (this.container.style.visibility != "visible"){
+            this.container.style.visibility = 'visible';
+            
+            this.createHealthBars();
+            this.updateHealthBar();
+
+            let updateData = () => { setTimeout(() => { this.updateHealthBar(); }, 50); }
+            this.on(this.wGame.dofus.connectionManager, 'GameFightTurnStartMessage',updateData);
+            this.on(this.wGame.dofus.connectionManager, 'GameFightTurnEndMessage',updateData);
+            this.on(this.wGame.gui,'GameActionFightLifePointsGainMessage',updateData);
+            this.on(this.wGame.gui,'GameActionFightLifePointsLostMessage',updateData);
+            this.on(this.wGame.gui,'GameActionFightLifeAndShieldPointsLostMessage',updateData);
+            this.on(this.wGame.gui,'GameActionFightPointsVariationMessage',updateData);
+            this.on(this.wGame.gui,'GameFightOptionStateUpdateMessage',updateData);
+            this.on(this.wGame.gui,'GameActionFightDeathMessage',updateData);
+            this.on(this.wGame.gui,'resize',updateData);
+        }
     }
 
-    private stopOnFightEnd(): void {
-        this.on(this.wGame.dofus.connectionManager, 'GameFightEndMessage', () => {
-            try {
-                this.barContainer.fightEnded();
-            } catch (ex) {
-                Logger.error(ex);
-            }
-        });
+    private updateHealthBar() {
+        this.createHealBar();
     }
 
-    private stopOnFightStop(): void {
-        this.on(this.wGame.dofus.connectionManager, 'GameFightLeaveMessage', () => {
-            try {
-                this.barContainer.fightEnded();
-            } catch (ex) {
-                Logger.error(ex);
+    private createHealBar(){
+        let fighters = this.wGame.gui.fightManager.getFighters();
+        for (let index in fighters) {
+            let fighter = this.wGame.gui.fightManager.getFighter(fighters[index]);
+            if (fighter.data.alive) {
+                if (!this.bars[fighter.id]) {
+                    this.bars[fighter.id] = new Bar(fighter, this.wGame); 
+                    this.addOnResetListener(() => { this.destoryHealthBars(); });
+                }
+                this.bars[fighter.id].update();
             }
-        });
+        }
     }
 
-
-    public reset() {
-        super.reset();
-        if (this.params.health_bar) {
-            if (this.shortcutsHelper) this.shortcutsHelper.unBindAll();
-            if (this.barContainer) this.barContainer.destroy();
-            const healthbarCss = this.wGame.document.getElementById('healthbarCss');
-            if (healthbarCss && healthbarCss.parentElement) healthbarCss.parentElement.removeChild(healthbarCss);
+    private destroyHealthBar(fighterId: any){
+        if (this.bars[fighterId]) {
+            this.bars[fighterId].destroy();
+            delete this.bars[fighterId];
```

Comment:
Delete removes the content of a memory reference, but not the memory reference itself. Instead, use `Array.prototype.splice`

Reply:
I don't have a lot of experience in this language, so do you mind writing the code i should use to replace "delete this.bars[fighterId];"? 
I wasn't able to fully understand your point...

Reply:
@Maryannah Isn't splice meant for Arrays? `this.bars` is an object here

Reply:
If so my bad then, ignore this comment !